certbot/certbot/_internal/main.py

```diff
@@ -1654,6 +1655,129 @@ def make_or_verify_needed_dirs(config: configuration.NamespaceConfig) -> None:
+    changes = orig_renewal_params_set ^ final_renewal_params_set
+
+    if len(changes) == 0:
+        success_message = '\nNo changes were made to the renewal configuration.'
+    else:
+        success_message = '\nSuccessfully updated configuration.' + \
+                          '\nChanges will apply when the certificate renews.'
+
+    display_util.notify(success_message)
+
+
+def reconfigure(config: configuration.NamespaceConfig,
+          plugins: plugins_disco.PluginsRegistry) -> None:
+    """Allow the user to set new configuration options for an existing certificate without
+       forcing renewal. This can be used for things like authenticator, installer, and hooks,
+       but not for the domains on the cert, since those are only saved in the cert.
+
+    :param config: Configuration object
+    :type config: configuration.NamespaceConfig
+
+    :param plugins: List of plugins
+    :type plugins: plugins_disco.PluginsRegistry
+
+    :raises errors.Error: if the dry run fails
+    :raises errors.ConfigurationError: if certificate could not be loaded
+
+    """
+
+    if config.domains:
+        raise errors.ConfigurationError("You have specified domains, but this function cannot "
+            "be used to modify the domains in a certificate. If you would like to do so, you "
+            "will need to renew the certificate instead of reconfiguring, which can be done "
+            "with the flag --force-renewal. Otherwise, remove the domains from the command "
+            "to continue reconfiguring. You can specify which certificate you want on the command "
+            "line with flag --cert-name instead.")
+    # While we could technically allow domains to be used to specify the certificate in addition to
+    # --cert-name, there's enough complexity with matching certs to domains that it's not worth it,
+    # to say nothing of the difficulty in explaining what exactly this subcommand can modify
+
+
+    # To make sure that the requested changes work, do a dry run. While setting up the dry run,
+    # we will set all the needed fields in config, which will then be saved upon success.
+    config.dry_run = True
+
+    if not config.certname:
+        certname_question = "Which certificate would you like to reconfigure?"
+        config.certname = cert_manager.get_certnames(
+            config, "reconfigure", allow_multiple=False,
+            custom_prompt=certname_question)[0]
+
+    certname = config.certname
+
+    try:
+        renewal_file = storage.renewal_file_for_certname(config, certname)
+    except errors.CertStorageError:
+        raise errors.ConfigurationError(f"An existing certificate with name {certname} could not "
+            "be found. Run `certbot certificates` to list available certificates.")
+
+    # figure this out before we modify config
+    if config.deploy_hook and not config.run_deploy_hook:
+        msg = "You are attempting to set a new deploy hook. Would you like Certbot to run the new "
```

Comment:
Very minor thing, if you like it: I've been in the habit of always using the flag name where possible, to make things more obvious to the user.

```suggestion
        msg = "You are attempting to set a new --deploy-hook. Would you like Certbot to run the new "
```

Reply:
love this!